src/components/ContactsApp.jsx

```diff
@@ -63,33 +49,19 @@ class ContactsApp extends React.Component {
 
     return (
       <main className="app-content">
-        <ContactsHeader />
+        <ContactsHeaderWithActions />
         <div role="contentinfo">
           <ContactsList
             onClickContact={this.displayContactCard}
             contacts={this.props.contacts}
           />
         </div>
         {displayedContact && (
-          <Modal
-            into="body"
-            dismissAction={this.hideContactCard}
-            size="xxlarge"
-          >
-            <ModalContent>
-              <TranslatedContactCard
-                contact={displayedContact}
-                renderActions={() => (
-                  <ContactCardMenu
-                    deleteAction={{
-                      label: this.context.t("delete"),
-                      action: () => this.onDeleteContact(displayedContact)
-                    }}
-                  />
-                )}
-              />
-            </ModalContent>
-          </Modal>
```

Comment:
Here I move the `<Moda />` wrapper to a wrapper component called `<ContactCardModal />`. It let the code be more expressive about responsibility and avoid clojure misusage: `<ContactCardModal />` takes 3 props:
1. hideModal: to let it hide the modal
1. onDeleteContact: to let it delete a contact
1. contact: to let it know about contact

Maybe it could be better to remove `onDeleteContact` from its props and prefer to give it a menu: then the modal will get a context (`contact`), a menu (actions that can be performed) and a `hideModal` prop that is mandatory.